haystack/components/converters/docx.py

```diff
@@ -69,21 +97,50 @@ class DOCXToDocument:
 
     Usage example:
     ```python
-    from haystack.components.converters.docx import DOCXToDocument
+    from haystack.components.converters.docx import DOCXToDocument, TableFormat
 
-    converter = DOCXToDocument()
+    converter = DOCXToDocument(table_format=TableFormat.CSV)
     results = converter.run(sources=["sample.docx"], meta={"date_added": datetime.now().isoformat()})
     documents = results["documents"]
     print(documents[0].content)
     # 'This is a text from the DOCX file.'
     ```
     """
 
-    def __init__(self):
+    def __init__(self, table_format: Union[str, TableFormat] = TableFormat.CSV):
         """
         Create a DOCXToDocument component.
+
+        :param table_format: The format for table output. Can be either TableFormat.MARKDOWN,
+            TableFormat.CSV, "markdown", or "csv". Defaults to TableFormat.CSV.
         """
         docx_import.check()
+        self.table_format = TableFormat.from_str(table_format) if isinstance(table_format, str) else table_format
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Serializes the component to a dictionary.
+
+        :returns:
+            Dictionary with serialized data.
+        """
+        return default_to_dict(self, table_format=str(self.table_format))
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "DOCXToDocument":
+        """
+        Deserializes the component from a dictionary.
+
+        :param data:
+            The dictionary to deserialize from.
+        :returns:
+            The deserialized component.
+        """
+        # Convert the table_format string back to enum before passing to the constructor
+        if "init_parameters" in data and "table_format" in data["init_parameters"]:
```

Comment:
Those two keys are always going to be present - we can remove this check.

Reply:
Whoops, this  `table_format` key won't be present in existing serialized pipelines - we should still check for that. Sorry about the confusion.